File: src/test/java/org/ybiquitous/messages/MessageKeyTest.java
package org.ybiquitous.messages;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.assertThat;

import java.util.Locale;

import org.junit.Test;

public class MessageKeyTest {

    @Test
    public void simple() {
        MessageKey test_key = newMessageKey("test.key");
        assertThat(test_key.get(1, 2, "三"), is("1 足す 2 は 三"));

        MessageKey test_key1 = newMessageKey("test.key1");
        MessageKey test_key2 = newMessageKey("test.key2");
        assertThat(test_key2.get(test_key1), is("'aaa' from other key"));
    }

    @Test
    public void locale() {
        MessageKey test_key = newMessageKey("test.key");
        assertThat(test_key.get(1, 2, "三"), is("1 足す 2 は 三"));

        MessageLocaleHolder.set(Locale.ENGLISH);
        assertThat(test_key.get(1, 2, 3), is("1 plus 2 is 3"));

        MessageLocaleHolder.reset();
        assertThat(test_key.get(1, 2, "三"), is("1 足す 2 は 三"));
        assertThat(test_key.get(Locale.ENGLISH, 1, 2, 3), is("1 plus 2 is 3"));
    }

    private MessageKey newMessageKey(String key) {
        return new MessageKey(key, "messages");
    }
}

File: src/test/scala/org/ybiquitous/messages/scala/MessageKeyTest.scala
package org.ybiquitous.messages.scala

import java.util.Locale

import org.hamcrest.CoreMatchers._
import org.junit.Test
import org.junit.Assert.assertThat

class MessageKeyTest {

  @Test def equality() {
    val key = MessageKey("test.key")
    val key2 = MessageKey("test.key")
    assertThat(key, is(key2))
  }

  @Test def simple() {
    val key = MessageKey("test.key")
    assertThat(key.get(1, 2, "3"), is("1 足す 2 は 3"))
  }

  @Test def locale() {
    val key = MessageKey("test.key")
    assertThat(key.get(Locale.ENGLISH, 1, 2, "3"), is("1 plus 2 is 3"))
  }
}
